File: src/lib/types/ui/recent/RecentFileListView.ts
export interface RecentFileListView {
  id: number;
  library_id: number | null;
  path: string;
  last_played_at: string; // ISO date string
  last_position: number;
  play_count: number;

  // Champs venant de library_cache (flat)
  cache_id: number | null;
  cache_path: string | null;
  title: string | null;
  artist: string | null;
  album: string | null;
  album_artist: string | null;
  year: string | null;
  genre: string | null;
  track_number: number | null;
  disc_number: number | null;
  duration: number | null;
  bitrate: number | null;
  bits_per_sample: number | null;
  sample_rate: number | null;
  channels: number | null;
  audio_format: string | null;
  mime_type: string | null;
  file_size: number | null;
  extra_tags: string | null;
  thumbnail_path: string | null;
  last_scanned_at: string | null;

  liked: boolean;
}

File: src-tauri/src/mapper/recent/recent_file_list_view.rs

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct RecentFileListView {
    pub id: i64,
    pub library_id: Option<i64>,
    pub path: String,
    pub last_played_at: DateTime<Utc>,
    pub last_position: f64,
    pub play_count: i64,

    // Champs venant de library_cache (flat)
    pub cache_id: Option<i64>,
    pub cache_path: Option<String>,
    pub title: Option<String>,
    pub artist: Option<String>,
    pub album: Option<String>,
    pub album_artist: Option<String>,
    pub year: Option<String>,
    pub genre: Option<String>,
    pub track_number: Option<i64>,
    pub disc_number: Option<i64>,
    pub duration: Option<f64>,
    pub bitrate: Option<i64>,
    pub bits_per_sample: Option<i64>,
    pub sample_rate: Option<i64>,
    pub channels: Option<i64>,
    pub audio_format: Option<String>,
    pub mime_type: Option<String>,
    pub file_size: Option<i64>,
    pub extra_tags: Option<String>,
    pub thumbnail_path: Option<String>,
    pub last_scanned_at: Option<DateTime<Utc>>,

    pub liked: bool,
}